### DISCUSSION TITLE
Instructions

### DISCUSSION THREAD

**Discussion by ghost** [farag2/Install-Office#4] [General]
Hey! Just stumbled upon your repo. I've been trying to find an offline installer for Microsoft Office 2021, and all Microsoft could give me is a link for Office 365, pretty frustrating!

How can I use the .ps1 file? I ran it multiple times, but nothing really happened.

Thanks for your help!

**Comment by farag2** [comment]
Hi! Just download the archive: https://github.com/farag2/Office/releases/tag/2.1.2
There is 2 important file: the script and default uncofigured xml. Expand these 2 files. Configure ps1. As you may see it supports downloading 2019, 2021 and 365 Office with the appropriate apps.
By default it downloads `DownloadOffice -Branch ProPlus2021Volume -Channel PerpetualVL2021 -Components Excel, Word`
Add the apps you need (see the examples). Save the file and run even with Powershell the context menu. It will start downloading the latest ODT, expand it, remove all unnecessary files and download Office. There will a console opened. No progress bar or any message. Near the ps1 file a new Office folder will appear. It will from 2 to 3,5 GB. Just wait and console will close itself. Now open the ps1 file again (do not touch xml: now it is configured) and place `#` before the  `DownloadOffice -Branch ProPlus2021Volume -Channel PerpetualVL2021 -Components Excel, Word` string. And uncomment `Start-Process -FilePath "$PSScriptRoot\setup.exe" -ArgumentList "/configure `"$PSScriptRoot\Config.xml`"" -Wait`. Save file and run it again ia right button click. It will start installing with a nice installer automatically without any buttons to choose or stop. As soon as it will install, everything will be closed. All apps will be installed. The Office folder might be removed. :)

**Comment by farag2** [comment]
For sure now you ask: Why this is so strange way to download and install it? All because it is built for admins, not an average users. There is no more old-school installers for Office like it was until Office 2016 with all components to choose from. Only pure config!

**Comment by ghost** [comment]
Thank you for such a detailed guide! This is incredibly helpful. Like you said, this is all about modifications and what one wants. I searched up things such as different branches (ProfessionalPlus vs Home & Business), components (Excel, PowerPoint, etc.), and I will make a custom installer with your guide. Thanks again!!

**Comment by farag2** [comment]
Read how ODT works: https://github.com/farag2/Office#links. Everything is documented in MS. 🐱

**Comment by Timmmmaaahh** [comment]
Hi! Thank you for this amazing script. I found that '.\Download.ps1 -Branch ProPlus2021Volume -Channel PerpetualVL2021 -Components Excel, Word, PowerPoint' downloads the 'Microsoft Office LTSC Professional Plus 2021' version. Is there a way to make it download the 'Microsoft Office Professional Plus 2021' (non-LTSC) version? I tried 'ProPlus2021Retail' but only the 2019 version is supported this way.

**Comment by farag2** [comment]
2021 is only LTSC version. :)

**Reply by Timmmmaaahh** [reply]
Thanks for your reply. Apparently, they are not the same 🤷‍♂️ See screenshot:
![mstsc_IOY6dWpTom](https://github.com/farag2/Office/assets/19148757/5b6c82f1-c58e-4a1b-91e5-39a54681280c)
I needed to uninstall that version and managed to install the non-LTSC via ODT using the following line: `<Product ID="ProPlus2021Volume" PIDKEY="My_Product_Key">`. Channel is also PerpetualVL2021, but I'm assuming it downloaded the correct version because I entered my product key in the configuration. In any case, it didn't download the same version. Microsoft probably doesn't even know what versions they all have 😅

**Reply by farag2** [reply]
MS knows. It's LTSC only. Because it has a long term support.
If you need to uninstall, use the script from the repo.

**Comment by Kalyan-Kumar-kk** [comment]
hey i'm stuck with this error saying unable to find specified config file.
office folder was downloaded and tried to install from context menu
<img width="463" alt="image" src="https://github.com/user-attachments/assets/caca9906-1418-4f51-87c8-187171f5f981" />

**Reply by farag2** [reply]
Hello. Please share with arguments you try to download Office.

**Reply by Kalyan-Kumar-kk** [reply]
latest  --- .\Download.ps1 -Branch ProPlus2019Retail -Channel Current -Components Word, Excel, PowerPoint

i have also tried with   .\Download.ps1 -Branch O365ProPlusRetail -Channel Current -Components Excel, OneDrive, Outlook, PowerPoint, Teams, Word

**Reply by farag2** [reply]
Did you remove XML file after downloading or move downloaded Office folder to another location?
Your error means that setup.exe cannot find XML configuration file.

**Reply by Kalyan-Kumar-kk** [reply]
no nothing i have not moved
<img width="590" alt="image" src="https://github.com/user-attachments/assets/9d0788fb-9394-4a8f-87b7-89b272bf7439" />

**Reply by farag2** [reply]
Could you provide a remote access via AnyDesk? Reach me via telegram: sanctuary_d

**Comment by ilhass** [comment]
It's really complicated and hard to understand, and nothing is installing or downloading. If only the guide were clearer

**Reply by farag2** [reply]
Please read the manual before proceeding. Which problem do you encounter with?

**Comment by Ishrak-Saleh** [comment]
Hello! I just wanted to know how safe is this. What server is this getting downloaded from?
If I connect my microsoft account will I face any issues?

**Reply by farag2** [reply]
Hello. all actions are done via ODT: https://github.com/farag2/Install-Office/blob/0fda1e0e260f8ed1483c07f78a02e9520de80215/Download.ps1#L284
Nothing is stored on an external recourses except Microsoft ones.

**Reply by Ishrak-Saleh** [reply]
Thank you for your work and such quick reply. I appreciate your efforts.

**Reply by farag2** [reply]
You're welcome. I wrote this script for myself))

**Comment by minautumn4Z** [comment]
i wait too long but the Office folder never show up... can u help me pls

**Reply by farag2** [reply]
Hm... that's strange. Probably, there's an issue with a connection to MS server. I can take a look remotely via RustDesk this evening. Please get in touch with me via Telegram, if you're OK. https://t.me/sanctuary_d

**Comment by minautumn4Z** [comment]
oh i ve done it last night. tysm. but i have another question. when i open these apps, there is alway pop-up a tab said i dont have key, is that a normal thing or i need to uninstall and install again?
![Untitled](https://github.com/user-attachments/assets/e7f30b9d-ba58-45fa-b141-102a0c3775f9)

**Comment by farag2** [comment]
Hello again. Of course it will ask you for a license. :D Why you thhought that this is a cracked Office?) use https://massgrave.dev/

**Reply by minautumn4Z** [reply]
tysm

**Comment by farag2** [comment]
Alos, where did you get 2016 version? It is not a C2R, but an MSI one. Where did you download it?

**Reply by minautumn4Z** [reply]
I downloaded from the install-office master in the post and following your guide

**Reply by farag2** [reply]
The script doesn't provide an option to download 2016 version. that's why I asked you.
